spec/scoreCardSpec.js

```diff
@@ -0,0 +1,131 @@
+describe('ScoreCard', function() {
+   var scoreCard;
+
+   beforeEach(function() {
+      scoreCard = new ScoreCard();
+   });
+
+   describe('Defaults', function() {
+      it('does not have frames by default', function() {
+         expect(scoreCard.frames.length).toEqual(0)
+      });
+   });
+
+   describe('Creating new frame', function() {
+      it('records score on a new frame when game starts', function() {
+         scoreCard.recordScore(1);
+
+         expect(scoreCard.frames.length).toEqual(1);
+         expect(scoreCard.frames[0].getTotalScore()).toEqual(1);
+      });
+
+      it('records score on a new frame when previous frame ended with a strike', 
+      function() {
+         scoreCard.recordScore(10);
+
+         expect(scoreCard.frames.length).toEqual(1);
+
+         scoreCard.recordScore(1);
+
+         expect(scoreCard.frames.length).toEqual(2);
+         expect(scoreCard.frames[1].getTotalScore()).toEqual(1);
+      });
+
+      it('records score on a new frame when previous frame ended after two rolls', 
+      function() {
+         scoreCard.recordScore(1);
+         scoreCard.recordScore(2);
+
+         expect(scoreCard.frames.length).toEqual(1);
+
+         scoreCard.recordScore(4);
+
+         expect(scoreCard.frames.length).toEqual(2);
+         expect(scoreCard.frames[1].getTotalScore()).toEqual(4);
```

Comment:
Is this tested in isolation? You may want to use doubles (or spies) here to not reply on Frame when testing ScoreCard.

Reply:
Or, maybe this test is trying to test a behaviour that belongs to Frame and not ScoreCard? If this is the case, then maybe both classes are too coupled. It would be interesting to see how the responsibilities are separated - maybe in a diagram.